packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -68,60 +67,59 @@ const GridVisualizerGrid = forwardRef(
 			};
 		}, [ gridElement ] );
 
-		useEffect( () => {
-			function onGlobalDrag() {
-				setIsDroppingAllowed( true );
-			}
-			function onGlobalDragEnd() {
-				setIsDroppingAllowed( false );
-			}
-			document.addEventListener( 'drag', onGlobalDrag );
-			document.addEventListener( 'dragend', onGlobalDragEnd );
-			return () => {
-				document.removeEventListener( 'drag', onGlobalDrag );
-				document.removeEventListener( 'dragend', onGlobalDragEnd );
-			};
-		}, [] );
-
 		return (
-			<BlockPopoverCover
-				className={ clsx( 'block-editor-grid-visualizer', {
-					'is-dropping-allowed': isDroppingAllowed,
-				} ) }
-				clientId={ gridClientId }
-				__unstablePopoverSlot="__unstable-block-tools-after"
-			>
-				<div
-					ref={ ref }
-					className="block-editor-grid-visualizer__grid"
-					style={ gridInfo.style }
-				>
-					{ isManualGrid ? (
-						<ManualGridVisualizer
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-						/>
-					) : (
-						Array.from( { length: gridInfo.numItems }, ( _, i ) => (
-							<GridVisualizerCell
-								key={ i }
-								color={ gridInfo.currentColor }
-							/>
-						) )
-					) }
-				</div>
-			</BlockPopoverCover>
+			<>
+				<GridPopunder
+					gridClientId={ gridClientId }
+					gridInfo={ gridInfo }
+					isManualGrid={ isManualGrid }
+				/>
+				{ isManualGrid && (
+					<GridPopover
+						ref={ ref }
+						gridClientId={ gridClientId }
+						gridInfo={ gridInfo }
+					/>
+				) }
+			</>
 		);
 	}
 );
 
-function ManualGridVisualizer( { gridClientId, gridInfo } ) {
+/**
+ * A popover component that renders in a slot over the grid block.
+ *
+ * This provides interactive elements of the grid visualization —
+ * block inserters and drop zones.
+ *
+ * @param {Object} props
+ * @param {string} props.gridClientId
+ * @param {Object} props.gridInfo
+ */
+const GridPopover = forwardRef( ( { gridClientId, gridInfo }, ref ) => {
```

Comment:
`ManualGridVisualizer` is now renamed to `GridPopover`, and it now encompasses all the interactive aspects of the grid visualizer. It renders it's own popover over the grid.